toolkit/tools/scheduler/schedulerutils/printresults.go

```diff
@@ -33,6 +34,99 @@ func PrintBuildResult(res *BuildResult) {
 	}
 }
 
+// RecordBuildSummary stores the summary in to a csv.
+func RecordBuildSummary(pkgGraph *pkggraph.PkgGraph, graphMutex *sync.RWMutex, buildState *GraphBuildState, outputPath string) {
+
+	graphMutex.RLock()
+	defer graphMutex.RUnlock()
+
+	failedSRPMs := make(map[string]*pkggraph.PkgNode)
+	failures := buildState.BuildFailures()
+	for _, failure := range failures {
+		failedSRPMs[failure.Node.SrpmPath] = failure.Node
+	}
+
+	prebuiltSRPMs := make(map[string]*pkggraph.PkgNode)
+	builtSRPMs := make(map[string]*pkggraph.PkgNode)
+	unbuiltSRPMs := make(map[string]*pkggraph.PkgNode)
+	unresolvedDependencies := make(map[string]bool)
+
+    buildNodes := pkgGraph.AllBuildNodes()
+    for _, node := range buildNodes {
+        if buildState.IsNodeCached(node) {
+            prebuiltSRPMs[node.SrpmPath] = node
+            continue
+        } else if buildState.IsNodeAvailable(node) {
+            builtSRPMs[node.SrpmPath] = node
+            continue
+        }
+
+		_, found := failedSRPMs[node.SrpmPath]
+		if !found {
+            unbuiltSRPMs[node.SrpmPath] = node
+        }
+    }
+
+	for _, node := range pkgGraph.AllRunNodes() {
+		if node.State == pkggraph.StateUnresolved {
+			unresolvedDependencies[node.VersionedPkg.String()] = true
+		}
+	}
+
+    csv_string := "Package,State,Blocker"
+
+	for srpm := range builtSRPMs {
+        csv_string += "\n" + filepath.Base(builtSRPMs[srpm].SrpmPath) + ",Built,,,"
+	}
+
+	for srpm := range prebuiltSRPMs {
+        csv_string += "\n" + filepath.Base(prebuiltSRPMs[srpm].SrpmPath) + ",PreBuilt,,,"
```

Comment:
Although it wont matter as an optimization, but an idiomatic approach would be to use a `Builder`.
Reference: [String Builder Example](https://pkg.go.dev/strings#example-Builder) & [String Builder](https://pkg.go.dev/strings#Builder)

Reply:
Even better, Go has a built in CSV stream module: https://pkg.go.dev/encoding/csv

You can just pass  it a file and it will stream the CSV data straight in so we don't even have to buffer strings.